### PR TITLE
Add multiple namespaces support

### PR DESCRIPTION
**Pull request by viveksyngh**
This commit adds multiple namespaces support for the functions.

Closes: #34 

Signed-off-by: Vivek Singh <vivekkmr45@yahoo.in>

### CODE REVIEW

**Comment by alexellis** [comment]
How have you tested it?

**Comment by viveksyngh** [comment]
Yes I have tested it on local using tester helper. 

```
go run cmd/tester/main.go
Handling connection for 8080
2019/09/25 14:02:53 Syncing topic map
2019/09/25 14:02:54 Invoke function: nodeinfo.fn
Handling connection for 8080
Handling connection for 8080
2019/09/25 14:02:54 Syncing topic map
2019/09/25 14:02:54 Invoke function: cows.openfaas-fn
2019/09/25 14:02:54 connector-sdk got result: [200] vm.powered.on => nodeinfo.fn (89) bytes
[200] vm.powered.on => nodeinfo.fn
Hostname: nodeinfo-6df94d649d-46cfl

Platform: linux
Arch: x64
CPU count: 4
Uptime: 9757

2019/09/25 14:02:54 tester got result: [200] vm.powered.on => nodeinfo.fn (89) bytes
2019/09/25 14:02:54 connector-sdk got result: [200] vm.powered.on => cows.openfaas-fn (105) bytes
[200] vm.powered.on => cows.openfaas-fn
          (__)
          (oo)
   /-------\/ 5+2=?
  / !     !!
 *  !!----!!
    ^^    ^^
cow who cow'nts

2019/09/25 14:02:54 tester got result: [200] vm.powered.on => cows.openfaas-fn (105) bytes
Handling connection for 8080
2019/09/25 14:02:55 Syncing topic map
Handling connection for 8080
2019/09/25 14:02:56 Syncing topic map
2019/09/25 14:02:56 Invoke function: nodeinfo.fn
Handling connection for 8080
2019/09/25 14:02:56 Invoke function: cows.openfaas-fn
2019/09/25 14:02:56 connector-sdk got result: [200] vm.powered.on => nodeinfo.fn (89) bytes
[200] vm.powered.on => nodeinfo.fn
Hostname: nodeinfo-6df94d649d-46cfl

Platform: linux
Arch: x64
CPU count: 4
Uptime: 9760

2019/09/25 14:02:56 tester got result: [200] vm.powered.on => nodeinfo.fn (89) bytes
2019/09/25 14:02:57 connector-sdk got result: [200] vm.powered.on => cows.openfaas-fn (134) bytes
[200] vm.powered.on => cows.openfaas-fn
               (__)
(__)           (oo)
(oo)      ______\/
-\/    /|      |
| |   * |______|
| |        ||
---        --
Cow Computer

2019/09/25 14:02:57 tester got result: [200] vm.powered.on => cows.openfaas-fn (134) bytes
Handling connection for 8080
2019/09/25 14:02:57 Syncing topic map
Handling connection for 8080
2019/09/25 14:02:58 Syncing topic map
2019/09/25 14:02:59 Invoke function: nodeinfo.fn
Handling connection for 8080
2019/09/25 14:02:59 Invoke function: cows.openfaas-fn
2019/09/25 14:02:59 connector-sdk got result: [200] vm.powered.on => nodeinfo.fn (89) bytes
[200] vm.powered.on => nodeinfo.fn
Hostname: nodeinfo-6df94d649d-46cfl

Platform: linux
Arch: x64
CPU count: 4
Uptime: 9762

2019/09/25 14:02:59 tester got result: [200] vm.powered.on => nodeinfo.fn (89) bytes
2019/09/25 14:02:59 connector-sdk got result: [200] vm.powered.on => cows.openfaas-fn (283) bytes
[200] vm.powered.on => cows.openfaas-fn
   ___________________________
| |                (__)       \
| |                (oo)        |
| |         /-------\/          \____
| |        / |     ||            ____)   Cow'ndom
| |       *  ||----||           /
| |          ~~    ~~          |
| |___________________________/

2019/09/25 14:02:59 tester got result: [200] vm.powered.on => cows.openfaas-fn (283) bytes
Handling connection for 8080
2019/09/25 14:02:59 Syncing topic map
Handling connection for 8080
2019/09/25 14:03:00 Syncing topic map
2019/09/25 14:03:01 Invoke function: nodeinfo.fn
Handling connection for 8080
2019/09/25 14:03:01 Invoke function: cows.openfaas-fn
2019/09/25 14:03:01 connector-sdk got result: [200] vm.powered.on => nodeinfo.fn (89) bytes
[200] vm.powered.on => nodeinfo.fn
Hostname: nodeinfo-6df94d649d-46cfl

Platform: linux
Arch: x64
CPU count: 4
Uptime: 9764

2019/09/25 14:03:01 tester got result: [200] vm.powered.on => nodeinfo.fn (89) bytes
2019/09/25 14:03:01 connector-sdk got result: [200] vm.powered.on => cows.openfaas-fn (91) bytes
[200] vm.powered.on => cows.openfaas-fn
         (__)
         (oo)
   __[_]__\/
 / |     ||
*  ||----||
   ^^    ^^
 Portable Cow

2019/09/25 14:03:01 tester got result: [200] vm.powered.on => cows.openfaas-fn (91) bytes
Handling connection for 8080
2019/09/25 14:03:01 Syncing topic map
```

**Comment by alexellis** [comment]
Thank you. Is it still WIP, or ready to merge/test?

File: types/function_list_builder.go [hunk 1]
```diff
@@ -23,21 +24,64 @@ type FunctionLookupBuilder struct {
 	TopicDelimiter string
 }
 
-// Build compiles a map of topic names and functions that have
-// advertised to receive messages on said topic
-func (s *FunctionLookupBuilder) Build() (map[string][]string, error) {
-	var err error
+//getNamespaces get openfaas namespaces
+func (s *FunctionLookupBuilder) getNamespaces() ([]string, error) {
```
Comment (martindekov) [inline types/function_list_builder.go]: Can we add unit test for this ? We can stub the JSON response.

File: types/function_list_builder.go [hunk 2]
```diff
@@ -50,17 +94,38 @@ func (s *FunctionLookupBuilder) Build() (map[string][]string, error) {
...
-	serviceMap := buildServiceMap(&functions, s.TopicDelimiter)
-
-	return serviceMap, err
+	return functions, nil
 }
 
-func buildServiceMap(functions *[]types.FunctionStatus, topicDelimiter string) map[string][]string {
+// Build compiles a map of topic names and functions that have
+// advertised to receive messages on said topic
+func (s *FunctionLookupBuilder) Build() (map[string][]string, error) {
```
Comment (martindekov) [inline types/function_list_builder.go]: It would be tricky to add unit test here but it might be a good idea.

**Comment by viveksyngh** [comment]
This is almost ready and tested but I have not added unit test for this. Will add unit test and push it. 

Thanks @alexellis  @martindekov

**Comment by viveksyngh** [comment]
I have added unit test and updated the existing unit tests.

File: types/function_list_builder.go (hunk 1 above)
Comment (viveksyngh) [reply types/function_list_builder.go]: Added unit tests.

File: types/function_list_builder.go (hunk 2 above)
Comment (viveksyngh) [reply types/function_list_builder.go]: Yes, Unit test was already there, I have updated it.

**Comment by alexellis** [comment]
@martindekov WDYT?

**Comment by martindekov** [comment]
Thanks for the added unit tests @viveksyngh, @alexellis I think the implementation is good, I will vendor the change in the kafka-connector and possibly the vcenter-connector and try to test them also :+1:

**Comment by alexellis** [comment]
Thank you 👍 there is a new release ready.

**Comment by alexellis** [comment]
@embano1

**Pull request merged**